packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -925,8 +925,66 @@ export class Stack extends Construct implements ITaggable {
    * - You are now free to remove the `bucket` resource from `producerStack`.
    * - Don't forget to remove the `exportValue()` call as well.
    * - Deploy again (this time only the `producerStack` will be changed -- the bucket will be deleted).
+   *
+   * @deprecated Use `exportListOrStringValue`
+   */
+  public exportValue(exportedValue: any, options: ExportValueOptions = {}): string {
+    const importExpr = this.exportListOrStringValue(exportedValue, options);
+    if (typeof importExpr !== 'string') {
+      throw new Error('Tried to export a list with `exportValue()`. Please use `exportListOrStringValue()` instead.');
+    }
+    return importExpr;
+  }
+
+  /**
+   * Create a CloudFormation Export for a value
+   *
+   * Returns a string or string list representing the corresponding `Fn.importValue()`
+   * expression for this Export. If the value is a string list then the export is wrapped
+   * with an `Fn::Join` and the import is wrapped with an `Fn::split`, since CloudFormation
+   * requires any exported value evaluate to a string. You can control the name for the export by
+   * passing the `name` option.
```

Comment:
I also don't like copying this giant block of text for the comment, but I've seen elsewhere in the code that we leave the comments on deprecated functions. This one is just particularly large.